### PR TITLE
Added a function to create a TLS configuration

### PR DESCRIPTION
**Pull request by juandspy**
# Description

This function is used in some internal repositories in order to create Sarama consumers and producers.

## Type of change

Please delete options that are not relevant.

- New feature (non-breaking change which adds functionality)
- This change requires a documentation update
- Unit tests (no changes in the code)
- Documentation update

## Testing steps

`go test -v ./...`

### CODE REVIEW

**Review by Bee-lee — APPROVED** [review]
OK

**Comment by joselsegura** [comment]
Not sure if it is very related to the resto of the `http` package. Maybe it would be better to create a new package if this function doesn't match in any other

File: http/tls_config.go [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+		Certificates: []tls.Certificate{},
+		MinVersion:   tls.VersionTLS12,
+	}
+
+	// Load CA cert
+	caCert, err := ioutil.ReadFile(filepath.Clean(certPath))
+	if err != nil {
+		return nil, err
+	}
+	caCertPool := x509.NewCertPool()
```
Comment (tisnik) [inline http/tls_config.go:27]: no checks there?

File: http/tls_config.go [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+	caCertPool.AppendCertsFromPEM(caCert)
```
Comment (tisnik) [inline http/tls_config.go:28]: dtto

File: http/tls_config.go (hunk 1 above)
Comment (juandspy) [reply http/tls_config.go:27]: what can I check there?

File: http/tls_config.go (hunk 2 above)
Comment (juandspy) [reply http/tls_config.go]: added, tyvm for realizing

File: http/tls_config.go (hunk 1 above)
Comment (tisnik) [reply http/tls_config.go:27]: IIRC this constructor returns pointer, so maybe check for `nil` to avoid NPE

Comment (juandspy) [reply http/tls_config.go:27]: Updated!

**Comment by juandspy** [comment]
> Not sure if it is very related to the resto of the `http` package. Maybe it would be better to create a new package if this function doesn't match in any other

@joselsegura shall I create a `tlsutil` package?

**Comment by joselsegura** [comment]
> > Not sure if it is very related to the resto of the `http` package. Maybe it would be better to create a new package if this function doesn't match in any other
> 
> @joselsegura shall I create a `tlsutil` package?

I like the name :-)

File: testdata/cert.pem [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+kxyXJDrvEgDPKTNdcT1wm7+i562NL1kupD73AgMBAAGjUzBRMB0GA1UdDgQWBBSp
+fG4Foem3RyvuQDkorxH6ZhRZ3zAfBgNVHSMEGDAWgBSpfG4Foem3RyvuQDkorxH6
+ZhRZ3zAPBgNVHRMBAf8EBTADAQH/MA0GCSqGSIb3DQEBCwUAA4IBAQCFNE6GW4+y
+wxXAXiNNcw4/F9ytTmwz9pBLN2KEpjfqk2d7iyDWJKwtChVjadU3L5dNV9UKOJM7
+n5OG/rJSh82gjnRlgWIiR09FPZQoBH3WiGehnSdOcJmkQUcnIPuXN2TkWVUqFZcy
+1y1vcule0mLKjGH+ejwqbv7LgkS/i/AlGpeqavEXFi5IEAlXZbyJ9MT2j1eygMmj
+XQ8WgQb1DzDlSGiOawdLAP4ULTIaVgx4SnH/4+g51CPH8Aj69eYPcsRMKdjUxAwn
+LbFzsderXnoHddvHoWSEsOX43kikoSlGjW3p/vpWXhNQJiH9eYCSSHmIDcl8CCNt
+6DLxiUiQf33S
+-----END CERTIFICATE-----
```
Comment (joselsegura) [inline testdata/cert.pem:21]: Probably @tisnik knows better, but probably this file should be included in a `.gitleaks.toml` file in the root directory of the repository. You can see an example in Smart Proxy.
That file is intended to provide an "allow list" for certificates/secrets exposed in repositories to a scanner of this kind of data leaks run by Red Hat.

**Review by joselsegura — APPROVED** [review]
Waiting for the renaming of the package and minor comments about a possible warning because the cert file

**Pull request merged**